src/node/internal/crypto_sign.ts

```diff
@@ -0,0 +1,441 @@
+  const rsaPadding = getIntOption('padding', options);
+  const pssSaltLength = getIntOption('saltLength', options);
+
+  // Options specific to (EC)DSA
+  const dsaSigEnc = getDSASignatureEncoding(options);
+
+  if (typeof signature === 'string') {
+    signature = Buffer.from(signature, encoding);
+  }
+
+  return this[kHandle].verify(
+    key,
+    signature,
+    rsaPadding,
+    pssSaltLength,
+    dsaSigEnc
+  );
+};
+
+export function createSign(algorithm: string, options: any) {
+  return new Sign(algorithm, options);
+}
+
+export function createVerify(algorithm: string, options: any) {
+  return new Verify(algorithm, options);
+}
+
+type SignCallback = (err: any, signature?: Buffer) => void;
+type VerifyCallback = (err: any, valid?: boolean) => void;
+
+export function sign(
+  algorithm: string | null | undefined,
+  data: BufferSource,
+  options: any,
+  callback?: SignCallback
+): Buffer | undefined {
+  if (algorithm != null) {
+    validateString(algorithm, 'algorithm');
+  } else {
+    algorithm = undefined;
+  }
+  if (!isArrayBufferView(data)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      'data',
+      ['Buffer', 'TypedArray', 'DataView'],
+      data
+    );
+  }
+
+  if (!options) {
+    throw new ERR_CRYPTO_SIGN_KEY_REQUIRED();
+  }
+
+  const key = getPrivateKey(options);
+
+  // Options specific to RSA
+  const rsaPadding = getIntOption('padding', options);
+  const pssSaltLength = getIntOption('saltLength', options);
+
+  // Options specific to (EC)DSA
+  const dsaSigEnc = getDSASignatureEncoding(options);
+
+  if (callback === undefined) {
+    return Buffer.from(
+      cryptoImpl.signOneShot(
+        key,
+        algorithm,
+        data,
+        rsaPadding,
+        pssSaltLength,
+        dsaSigEnc
+      )
+    );
+  }
+
+  try {
+    callback(
+      null,
+      Buffer.from(
+        cryptoImpl.signOneShot(
+          key,
+          algorithm,
+          data,
+          rsaPadding,
+          pssSaltLength,
+          dsaSigEnc
+        )
+      )
+    );
+  } catch (err) {
+    queueMicrotask(() => callback(err));
+  }
+  return; // explicit return is necessary to squelch typescript warning.
+}
+
+export function verify(
+  algorithm: string | null | undefined,
+  data: BufferSource,
+  options: any,
+  signature: BufferSource,
+  callback?: VerifyCallback
+): boolean | undefined {
+  if (algorithm != null) {
+    validateString(algorithm, 'algorithm');
+  } else {
+    algorithm = undefined;
+  }
+
+  if (!isArrayBufferView(data)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      'data',
+      ['Buffer', 'TypedArray', 'DataView'],
+      data
+    );
+  }
+
+  if (!isArrayBufferView(signature)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      'signature',
+      ['Buffer', 'TypedArray', 'DataView'],
+      signature
+    );
+  }
+
+  if (!options) {
+    throw new ERR_CRYPTO_SIGN_KEY_REQUIRED();
+  }
+
+  let key: CryptoKey;
+  if (options instanceof CryptoKey) {
+    key = options as CryptoKey;
```

Comment:
Since you have the instanceof check, this is not needed

```suggestion
    key = options;
```

Reply:
It also does not hurt tho right?